matter_server/server/__main__.py

```diff
@@ -9,12 +10,41 @@
 from .matter_stack import MatterStack
 from .server import MatterServer
 
-logging.basicConfig(level=logging.WARN)
 _LOGGER = logging.getLogger(__package__)
-_LOGGER.setLevel(logging.DEBUG)
+
+
+def get_arguments() -> argparse.Namespace:
+    """Get parsed passed in arguments."""
+
+    parser = argparse.ArgumentParser(
+        description=" Matter Controller Server using WebSockets."
+    )
+    parser.add_argument(
+        "--log-file",
+        type=str,
+        default=None,
+        help="Log file to write to.  If not set, matter_server.log is used",
+    )
+    parser.add_argument(
+        "-log",
+        "--log-level",
+        default="info",
+        help="Provide logging level. Example --log-level debug, default=info, possible=(critical, error, warning, info, debug)",
+    )
+
+    arguments = parser.parse_args()
+
+    return arguments
 
 
 def main() -> int:
+    args = get_arguments()
+
+    if args.log_file:
+        file_handler = logging.FileHandler(args.log_file)
+
+    logging.basicConfig(handlers=[file_handler], level=args.log_level.upper())
```

Comment:
This currently calls `basicConfig` with unassigned `file_handler`. We should set handlers to None to let basicConfig create the default `StreamHandler` by default, e.g.

```suggestion
    handlers = None

    if args.log_file:
        handlers = [logging.FileHandler(args.log_file)]

    logging.basicConfig(handlers=handlers, level=args.log_level.upper())
```

Reply:
right! Missed that